src/contexts/TokenData.js

```diff
@@ -278,21 +279,109 @@ export function useTokenChartData(tokenAddress) { }
  * @param {*} timeWindow // a preset time window from constant - how far back to look
  * @param {*} interval  // the chunk size in seconds - default is 1 hour of 3600s
  */
-export function useTokenPriceData(tokenAddress, timeWindow, interval = 3600) { }
+export function useTokenPriceData(tokenAddress, timeWindow, interval = 3600) {}
 
 export function useAllTokenData() {
   const [state] = useTokenDataContext()
   const [stakingEnabled] = useStakingManager()
   const [pool2Enabled] = usePool2Manager()
   if (stakingEnabled || pool2Enabled) {
     // TODO Optimize
-    return Object.fromEntries(Object.entries(state).map(entry => [
-      entry[0],
-      {
-        ...entry[1],
-        tvl: entry[1].tvl + (stakingEnabled ? (entry[1].staking ?? 0) : 0) + (pool2Enabled ? (entry[1].pool2 ?? 0) : 0)
-      }
-    ]))
+    return Object.fromEntries(
+      Object.entries(state).map(entry => [
+        entry[0],
+        {
+          ...entry[1],
+          tvl: entry[1].tvl + (stakingEnabled ? entry[1].staking ?? 0 : 0) + (pool2Enabled ? entry[1].pool2 ?? 0 : 0)
+        }
+      ])
+    )
   }
   return state
 }
+
+export const useFilteredTokenData = ({ selectedChain = 'All', category = '' }) => {
+  const allChains = selectedChain === 'All'
+  const [allTokenData] = useTokenDataContext()
+
+  let totalStaking = 0
+  const [stakingEnabled] = useStakingManager()
+  let totalPool2 = 0
+  const [pool2Enabled] = usePool2Manager()
+
+  // All chains accumulated from priority list and from looping through tokens' specific chains
+  const chainsSet = new Set(priorityChainFilters)
+
+  let filteredTokens = Object.values(allTokenData).reduce((accTokenList, currTokenData) => {
+    // Skip all chain tokens and if there is a category, skip all tokens that are not in that category
+    if (
+      currTokenData.category === 'Chain' ||
```

Comment:
could just completely remove chain tokens from backend api now that we are switching frontend to a new one